src/service.ts

```
@@ -449,6 +880,407 @@ export class SolanaService extends Service {
+        const { amount, slippage } = await this.calculateOptimalBuyAmount(
+          initialQuote.inputMint,
+          initialQuote.outputMint,
+          wallet.amount
+        );
+        */
+        // amount is in atomic units (input token)
+        //
+        console.log('adjusted amount', amount, 'price impact slippage', slippage)
+        // adjust amount in initialQuote
+        initialQuote.inAmount = "" + amount // in input atomic units
+        delete initialQuote.swapUsdValue // invalidate
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)